src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -350,15 +349,26 @@ public ReferenceCounted touch(Object o) {
       Context ctx = Context.current();
       ProgressiveBackoff progressiveBackoff = new ProgressiveBackoff(retrier::newBackoff);
       AtomicLong committedOffset = new AtomicLong(0);
+
+      ListenableFuture<Void> callFuture = callAndQueryOnFailureWithRetrier(ctx, committedOffset, progressiveBackoff);
+
+      callFuture = Futures.catchingAsync(
+          callFuture,
+          Exception.class,
+          (e) -> {
+            Status status = Status.fromThrowable(e);
+            if (status != null && status.getCode() == Code.UNAUTHENTICATED) {
```

Comment:
I'd also check for Code.PERMISSION_DENIED here.

Reply:
I double-checked the documentation, and I'm not sure in which cases a service will return UNAUTHENTICATED vs. PERMISSION_DENIED. It seems like it won't hurt to also check PERMISSION_DENIED here, just in case?

Reply:
Done.